src/db/mongodb_repo.rs

```diff
@@ -50,4 +51,13 @@ impl MongoRepo {
             .await?
             .ok_or(Error::TaskNotFound)
     }
+
+    pub async fn get_all_todos(&self) -> Result<Vec<Todo>, Error> {
+        let cursor = self
+            .todos
+            .find(None, None)
+            .await?;
+        let v = cursor.try_collect().await?;
+        Ok(v)
+    }
```

Comment:
An alternative implementation that removes extra variables and result unwrapping/wrapping at the end:
`    pub async fn get_all_todos(&self) -> Result<Vec<Todo>, Error> {
        self
            .todos
            .find(None, None)
            .await?
            .try_collect::<Vec<Todo>>()
            .await
            .map_err(Error::from)
    }
`

Reply:
Brilliant suggestion, manages to get all the operations in one "chain", which is something I attempted to do multiple times, but failed.